.github/workflows/tests.yml

```diff
@@ -1,43 +1,49 @@
-name: tests
+name: Tests
 
 on:
-  push: null
-  pull_request: null
+  pull_request:
+  push:
+    branches: [master]
 
 jobs:
-  tests:
-    name: tests
-    runs-on: "ubuntu-latest"
+  run:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
+        os: [windows-latest, ubuntu-latest, macos-latest]
+      fail-fast: false
+
     steps:
       - name: cancel previous runs
-        uses: styfle/cancel-workflow-action@0.6.0
+        uses: styfle/cancel-workflow-action@0.10.0
         with:
           access_token: ${{ github.token }}
 
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
-      - uses: conda-incubator/setup-miniconda@v2
+      - name: Setup Micromamba
+        uses: mamba-org/provision-with-micromamba@main
         with:
-          python-version: 3.8
-          channels: conda-forge,defaults
-          channel-priority: strict
-          show-channel-urls: true
-
-      - name: configure conda and install code
+         environment-file: false
```

Comment:
[Seems](https://github.com/mamba-org/provision-with-micromamba#environment-name) that you should add `environment-name` in the case of `environment-file: false`?

Reply:
That is optional: https://github.com/mamba-org/provision-with-micromamba/issues/54

Reply:
Thanks for sharing that